### PR TITLE
Shopping List rendering for user

### LINKED ISSUE #2: 2. As a user, I want to read all of my shopping lists and the items in them.
## Summary

The most important feature of our app is the ability to look at their shopping lists to see what's on them, so the user knows what they need to buy. For this to work, we'll have to be able to choose a our shopping lists. The `List` (the active shopping list) and `Home` (where the user will be able to switch shopping lists) components are the best place in our code to do that. In fact, this feature is already partially implemented. Refer to the acceptance criteria in this issue, and the existing code, to understand the work that remains.

The `List` component is rendered on the `/list` page. Because the navigation links are currently broken, you'll have to type that path into your browser's address bar.

## Acceptance criteria
- [x] The `useShoppingLists` hook is used in `App.jsx` to get the shopping lists a user has access to from the database
- [x] The lists are passed into the `Home` component as a prop named `data`
- [x] The `setListPath` function is passed into the `Home` component so we can update the current list
- [x] The `useShoppingListData` hook is used in `App.jsx` to get the items in the current list from the Firestore database
- [x] The list is passed into the `List` component as a prop named `data`
- [x] In `Home.jsx`, the `SingleList` component is used to render the name of each shopping list a user is subscribed to
- [x] In `List.jsx`, the `ListItem` component is used to render the name of each item

## Notes
- ⚠️ Your **only goal** is to make sure that the names of the items render as expected. You might be tempted to modify `ListItem.jsx` or `SingleList.jsx` to show more information – you will do that **later**.
- The variables you need are already being passed into the `List` and `Home` components. What steps do you need to take in order to render them?
- Additionally, until issues [4](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/issues/4) and [5](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/issues/5) you should only be able to see the shopping list that was manually added in [issue 1](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/issues/1).

### PR DESCRIPTION
**Pull request by sdmisra**
## Description

This enables a user to view the shared lists for their email as well as the list items of those shared lists. 

## Related Issue

closes #2 

## Acceptance Criteria

- [x] The `useShoppingLists` hook is used in `App.jsx` to get the shopping lists a user has access to from the database
- [x] The lists are passed into the `Home` component as a prop named `data`
- [x] The `setListPath` function is passed into the `Home` component so we can update the current list
- [x] The `useShoppingListData` hook is used in `App.jsx` to get the items in the current list from the Firestore database
- [x] The list is passed into the `List` component as a prop named `data`
- [x] In `Home.jsx`, the `SingleList` component is used to render the name of each shopping list a user is subscribed to
- [x] In `List.jsx`, the `ListItem` component is used to render the name of each item

## Type of Changes

`new feature`

### Before
Screenshot 2024-02-07 at 12 29 02 PM
Screenshot 2024-02-07 at 12 29 53 PM

### After
Screenshot 2024-02-07 at 12 30 13 PM
Screenshot 2024-02-07 at 12 30 49 PM

## Testing Steps / QA Criteria

1. Review preview deployment that is based off this PR:
2. When on home page - sign in, review if 'dinner' list appears (shared list)
3. Click dinner to ensure local storage retrieves database info
4. navigate to [deployment address]/list
5. confirm that 'milk' and 'eggs' are visible within dinner list route

### CODE REVIEW

**Review by andiedoescode — COMMENTED** [review]
It all looks good!
Do you think the console logs in App.jsx (Ln 32, 33, 40) and firebase.js (Ln 33-35) will be helpful as we work with the API later on?

**Review by DevinaG007 — APPROVED** [review]
Aside from the commented lines Andrea mentioned, I think it looks good!

**Comment by sdmisra** [comment]
> It all looks good! Do you think the console logs in App.jsx (Ln 32, 33, 40) and firebase.js (Ln 33-35) will be helpful as we work with the API later on?

I think they might be helpful, but we could always add them as we need to in the future if we wanted to remove them for this PR. They were great for troubleshooting our initial attempts at using the Firebase data.

**Review by andiedoescode — APPROVED** [review]
Looks good! :D

File: src/App.jsx [hunk 1]
```diff
@@ -29,19 +29,20 @@ export function App() {
 	const { user } = useAuth();
 	const userId = user?.uid;
 	const userEmail = user?.email;
-
+	// console.log('userID', userId);
+	// console.log('email', userEmail)
```
Comment (cafloyd) [inline src/App.jsx]: I would echo Andrea and Devina's suggestion to remove these commented out console logs to clean up our files a bit!

```suggestion
```

File: src/App.jsx [hunk 2]
```diff
@@ -29,19 +29,20 @@ export function App() {
...
 	/**
 	 * This custom hook takes a user ID and email and fetches
 	 * the shopping lists that the user has access to.
 	 * Check ./api/firestore.js for its implementation.
 	 */
 	const lists = useShoppingLists(userId, userEmail);
+	// console.log('lists', lists);
```
Comment (cafloyd) [inline src/App.jsx]:
```suggestion
```

File: src/api/firebase.js [hunk 3]
```diff
@@ -30,7 +30,9 @@ export function useShoppingLists(userId, userEmail) {
 
 		// When we get a userEmail, we use it to subscribe to real-time updates
 		const userDocRef = doc(db, 'users', userEmail);
-
+		// console.log('database', db)
+		// console.log('email', userEmail)
+		// console.log('id', userId)
```
Comment (cafloyd) [inline src/api/firebase.js]:
```suggestion
```

**Review by cafloyd — CHANGES_REQUESTED** [review]
Really nice job sticking with what turned out to be a tougher issue this week, @sdmisra and @etiry !! 

Just one request to remove the commented out console logs left over from debugging! I think they got left in the hubub of troubleshooting during office hours 😊

**Comment by sdmisra** [comment]
Suggested changes from @andiedoescode, @DevinaG007, @cafloyd applied using the GitHub UI, I'll keep in mind that we don't want to leave in console logs in general when merging to main in the future

**Review by cafloyd — APPROVED** [review]
Looks great!! Thanks for cleaning things up, and really nice job troubleshooting those Firebase issues - I was really impressed to see your debugging process!

Congrats on completing your first issue, @sdmisra and @etiry !! You should be ok to merge this PR now!

**Pull request merged**